Fixes: Resetting Input Field In Organisation Tab

Sulochan-khadka opened the pull request:
## Proposed Changes

- Fixes #10783 
- Added SearchByMultipleFields

@ohcnetwork/care-fe-code-reviewers

https://github.com/user-attachments/assets/417cb944-3fa3-4269-9a94-9c7884e442a0


## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [ ] Completion of QA in Mobile Devices
- [ ] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
	- Enhanced search functionality in the Organization view, now utilizing a more efficient query management system.
	- Pagination limit set to 15, improving the browsing experience.
- **Refactor**
	- Simplified the logic for search and pagination, resulting in improved clarity and efficiency in the Organization view.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Sulochan-khadka committed:
Fixes: Resetting-input-field

Jacobjeevan commented:
- Follow the merge checklist please (requested the same in the last PR for this issue)
- Rithvik provided some suggestions there which was never replied to/implemented.

Sulochan-khadka commented:
> * Follow the merge checklist please (requested the same in the last PR for this issue)
> * Rithvik provided some suggestions there which was never replied to/implemented.

@Jacobjeevan 
1. Oh, i missed that. Will do it now
2. I implemented the changes in this PR, i guess. Since that PR was closed By @nihal467 as the ETA had passed, i had to create a new PR

rithviknishad commented:
Switching from input to search by multiple fields is not the solution for fixing the reset.

`SearchByMultipleField` is meant to be used for "searching with multiple fields" as the name suggests. what's the point in searching by single field? 🤔

Sulochan-khadka commented:
I guess i am unable to understand the issue here. @rithviknishad i also implemented  the same at first. But acting upon the suggestion of @Jacobjeevan that we should rather have SearchByMultipleFields , for consistency across the pages, made a commit https://github.com/ohcnetwork/care_fe/pull/10797/commits/f08e8fe16e622aa9aa578ebaa968eba458376e6a

which is why i moved forward with that. 
What do both of you suggest ? I do not want @nihal467 to close this PR as well 😕

Jacobjeevan commented:
> I guess i am unable to understand the issue here. @rithviknishad i also implemented the same at first. But acting upon the suggestion of @Jacobjeevan that we should rather have SearchByMultipleFields , for consistency across the pages, made a commit [f7413f4](https://github.com/ohcnetwork/care_fe/commit/f7413f4734eb3c7bf3c75866411954ee0bddefd9) which is why i moved forward with that. What do both of you suggest ? I do not want @nihal467 to close this PR as well 😕

Try to understand how the code works

(1) https://github.com/ohcnetwork/care_fe/pull/10797#discussion_r1978575429

In this case, the handleSearch isn't necessary, you can just use updateQuery instead.

(2) handleFieldChange isn't necessary since there's only one field anyways.

Sulochan-khadka committed:
removed-handleSearch-handleFieldChange

Sulochan-khadka commented:
> > I guess i am unable to understand the issue here. @rithviknishad i also implemented the same at first. But acting upon the suggestion of @Jacobjeevan that we should rather have SearchByMultipleFields , for consistency across the pages, made a commit [f7413f4](https://github.com/ohcnetwork/care_fe/commit/f7413f4734eb3c7bf3c75866411954ee0bddefd9) which is why i moved forward with that. What do both of you suggest ? I do not want @nihal467 to close this PR as well 😕
> 
> Try to understand how the code works
> 
> (1) [#10797 (comment)](https://github.com/ohcnetwork/care_fe/pull/10797#discussion_r1978575429)
> 
> In this case, the handleSearch isn't necessary, you can just use updateQuery instead.
> 
> (2) [#10797 (comment)](https://github.com/ohcnetwork/care_fe/pull/10797#discussion_r1978575988)
> 
> handleFieldChange isn't necessary since there's only one field anyways.

Understood...
Kindly check the latest commit.

Jacobjeevan committed:
adjusting styling for search

rithviknishad commented:
"Completion of QA" was not even conducted; This doesn't even work! 

https://github.com/user-attachments/assets/d04e53f7-7603-461a-88c4-2810b4365206
